frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts

```diff
@@ -87,6 +89,8 @@ export type GroupVersionKind = string;
  */
 export type K8sResourceKindReference = GroupVersionKind | string;
 
+export type K8sGroupVersionKind = { group: string; version: string; kind: string };
```

Comment:
The "right" thing to do here would be to make `group` optional and leave it off for core resources (instead of using group `"core"`). I don't know how big a change that is to existing code, however. We'd need to change utilities like `getGroupVersionKindForResource`.

This will also keep console from breaking if someone creates a CRD with API group `core`.

Reply:
making group optional make sense , will update.

I have doubt with another part i.e say for `getGroupVersionKindForResource` will return group only if the resource/model has it else group property won't be there as in below

```
export const getGroupVersionKindForResource: GetGroupVersionKindForResource = (resource) => {
  const { apiVersion, kind } = resource;
  const apiVersionSplit = apiVersion.split('/');
  const apiVersionSplitLen = apiVersionSplit.length;
  if (apiVersionSplitLen > 2) throw new Error('Provided resource has invalid apiVersion.');

  return {
    ...(apiVersionSplitLen === 2 && {
      group: apiVersionSplit[0],
    }),
    version: apiVersionSplitLen === 2 ? apiVersionSplit[1] : apiVersion,
    kind,
  };
};
```

Reply:
I tried it in case of false it does not add group even with `&&` 

```
let a = 1;
let b = {
  ...(a == 3 && {a}),
  b: 1
}

console.log(b) // { b: 1}
```